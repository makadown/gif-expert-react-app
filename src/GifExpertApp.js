import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ( {defaultCategories = ['Berserk']} ) => {
    const [categories, setCategories] = useState(defaultCategories);

    /*const handleAdd = () => {
        // setCategories([...categories, 'HunterXHunter']);
        setCategories( cats => [...cats, 'HunterXHunter']);
    }*/

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory agregarPapu={setCategories}></AddCategory>
            <hr/>
            <ol>
                {
                    categories.map( (c, i ) => {
                        return <GifGrid 
                            key={i} category={c}
                        />
                    })
                }
            </ol>
        </>
    )
}

export default GifExpertApp;