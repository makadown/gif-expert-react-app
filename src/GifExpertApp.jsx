import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
    
    // NEVER, NEVER, NEVER, Never create state conditionally. NEVER!
    const [categories, setCategories] = useState(['Donghua', 'Link Click', `One Piece`]);

    const onAddCategory = (newCategory) => {
        
        if (newCategory.trim().length === 0) return false;
        
        // serching categories case insensitive
        const foundCategory = categories.find(cat => cat.toLowerCase() === newCategory.toLowerCase());
        if (foundCategory) return false;

        // this adds the new category at the beginning of the array
        setCategories([newCategory, ...categories]);
        return true;
    }
    
    return ( <>
              
              <h1>Gif Expert App</h1>
              
              <AddCategory 
                            onAddCategory={onAddCategory}
                    />
              
            
              {
                  /* GIF ITEM */
                  categories.map( (category) => (
                        (
                          <GifGrid 
                            key={category}
                            category={category}
                          />
                        )
                    ) 
                  ) // map
              }                
              
    </> )
    ;
}