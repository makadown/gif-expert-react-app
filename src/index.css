@import url('https://fonts.googleapis.com/css2?family=Ubuntu&display=swap');

* {
  font-family: Ubuntu, Arial, Helvetica, sans-serif;
}

body {
  padding: 30px;
}

input {
  color:rgb(43, 42, 42);
  width: 100%;
  font-size: 1.1rem;
}

.card-grid {
  display:flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.card {
  align-content: center;
  border: 1px solid grey;
  border-radius: 6px;
  margin-bottom: 10px;
  margin-right: 10px;
  overflow: hidden;
}

.card p {
  padding: 5px;
  text-align: center;
}

.card img {
  max-height: 170px;
}