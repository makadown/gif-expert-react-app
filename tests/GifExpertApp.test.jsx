import {render, screen} from '@testing-library/react';
import {GifExpertApp} from '../src/GifExpertApp';

describe('GifExpertApp tests', () => {

    test('should match the snapshot', () => {
        const {container} = render(<GifExpertApp/>);        
        expect(container).toMatchSnapshot();
    })

    test('should show the title', () => {
        render(<GifExpertApp/>);
        expect(screen.getByText('Gif Expert App')).toBeTruthy();
    })

    test('should show the One Piece category by default', () => {
        render(<GifExpertApp/>);
        expect(screen.getByText('One Piece')).toBeTruthy();        
    })
})