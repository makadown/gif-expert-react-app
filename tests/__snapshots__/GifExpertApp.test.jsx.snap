// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp tests should match the snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form>
    <input
      id="myCategory"
      name="myCategory"
      placeholder="Search gifs"
      style="width: 25%; height: 10px;"
      type="text"
      value=""
    />
  </form>
  <h2>
    cargando....
  </h2>
  <h3>
    One Piece
     
  </h3>
  <div
    class="card-grid"
  />
</div>
`;
